Test case `org.apache.shardingsphere.elasticjob.lite.integrate.assertion.enable.oneoff.dataflow.StreamingDataflowElasticJobTest` may  block or throw exception sometimes

Sometimes the test is block because the `StreamingDataflowElasticJob.isCompleted()` is never return true which maybe by thread safe reason.

The exception will throw sometimes, exception is:

```
[ERROR] 2020-07-03 19:20:51,443 --Curator-TreeCache-0-- [org.apache.curator.framework.imps.CuratorFrameworkImpl] Ensure path threw exception 
java.lang.InterruptedException: null
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedNanos(AbstractQueuedSynchronizer.java:1039)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.tryAcquireSharedNanos(AbstractQueuedSynchronizer.java:1328)
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:277)
	at org.apache.curator.CuratorZookeeperClient.internalBlockUntilConnectedOrTimedOut(CuratorZookeeperClient.java:325)
	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:106)
	at org.apache.curator.framework.imps.NamespaceImpl.fixForNamespace(NamespaceImpl.java:83)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.fixForNamespace(CuratorFrameworkImpl.java:594)
	at org.apache.curator.framework.imps.ExistsBuilderImpl.forPath(ExistsBuilderImpl.java:151)
	at org.apache.curator.framework.imps.ExistsBuilderImpl.forPath(ExistsBuilderImpl.java:39)
	at org.apache.shardingsphere.elasticjob.lite.reg.zookeeper.ZookeeperRegistryCenter.isExisted(ZookeeperRegistryCenter.java:197)
	at org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodeStorage.isJobNodeExisted(JobNodeStorage.java:56)
	at org.apache.shardingsphere.elasticjob.lite.internal.election.LeaderService.hasLeader(LeaderService.java:91)
	at org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManager$LeaderElectionJobListener.isActiveElection(ElectionListenerManager.java:69)
	at org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManager$LeaderElectionJobListener.dataChanged(ElectionListenerManager.java:63)
	at org.apache.shardingsphere.elasticjob.lite.internal.listener.AbstractJobListener.childEvent(AbstractJobListener.java:42)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:732)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:726)
	at org.apache.curator.framework.listen.ListenerContainer$1.run(ListenerContainer.java:93)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:299)
	at org.apache.curator.framework.listen.ListenerContainer.forEach(ListenerContainer.java:85)
	at org.apache.curator.framework.recipes.cache.TreeCache.callListeners(TreeCache.java:725)
	at org.apache.curator.framework.recipes.cache.TreeCache.access$1400(TreeCache.java:71)
	at org.apache.curator.framework.recipes.cache.TreeCache$4.run(TreeCache.java:843)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[ERROR] 2020-07-03 19:20:51,448 --Curator-TreeCache-0-- [org.apache.curator.framework.recipes.cache.TreeCache]  
java.lang.NullPointerException: null
	at org.apache.shardingsphere.elasticjob.lite.internal.server.ServerNode.isLocalServerPath(ServerNode.java:61)
	at org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManager$LeaderElectionJobListener.isLocalServerEnabled(ElectionListenerManager.java:81)
	at org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManager$LeaderElectionJobListener.isActiveElection(ElectionListenerManager.java:69)
	at org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManager$LeaderElectionJobListener.dataChanged(ElectionListenerManager.java:63)
	at org.apache.shardingsphere.elasticjob.lite.internal.listener.AbstractJobListener.childEvent(AbstractJobListener.java:42)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:732)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:726)
	at org.apache.curator.framework.listen.ListenerContainer$1.run(ListenerContainer.java:93)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:299)
	at org.apache.curator.framework.listen.ListenerContainer.forEach(ListenerContainer.java:85)
	at org.apache.curator.framework.recipes.cache.TreeCache.callListeners(TreeCache.java:725)
	at org.apache.curator.framework.recipes.cache.TreeCache.access$1400(TreeCache.java:71)
	at org.apache.curator.framework.recipes.cache.TreeCache$4.run(TreeCache.java:843)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/election/ElectionListenerManager.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/election/ElectionListenerManager.java
@@ -18,6 +18,7 @@
 package org.apache.shardingsphere.elasticjob.lite.internal.election;
 
 import org.apache.curator.framework.recipes.cache.TreeCacheEvent.Type;
+import org.apache.shardingsphere.elasticjob.lite.handler.sharding.JobInstance;
 import org.apache.shardingsphere.elasticjob.lite.internal.listener.AbstractJobListener;
 import org.apache.shardingsphere.elasticjob.lite.internal.listener.AbstractListenerManager;
 import org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobRegistry;
@@ -25,6 +26,8 @@ import org.apache.shardingsphere.elasticjob.lite.internal.server.ServerNode;
 import org.apache.shardingsphere.elasticjob.lite.internal.server.ServerService;
 import org.apache.shardingsphere.elasticjob.lite.internal.server.ServerStatus;
 import org.apache.shardingsphere.elasticjob.lite.reg.base.CoordinatorRegistryCenter;
+
+import java.util.Objects;
 
 /**
  * Election listener manager.
@@ -70,7 +73,11 @@ public final class ElectionListenerManager extends AbstractListenerManager {
         }
         
         private boolean isPassiveElection(final String path, final Type eventType) {
-            return isLeaderCrashed(path, eventType) && serverService.isAvailableServer(JobRegistry.getInstance().getJobInstance(jobName).getIp());
+            JobInstance jobInstance = JobRegistry.getInstance().getJobInstance(jobName);
+            if (Objects.isNull(jobInstance)) {
+                return false;
+            }
+            return isLeaderCrashed(path, eventType) && serverService.isAvailableServer(jobInstance.getIp());
         }
         
         private boolean isLeaderCrashed(final String path, final Type eventType) {

--- a/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/server/ServerNode.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/internal/server/ServerNode.java
@@ -17,10 +17,12 @@
 
 package org.apache.shardingsphere.elasticjob.lite.internal.server;
 
+import org.apache.shardingsphere.elasticjob.lite.handler.sharding.JobInstance;
 import org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobRegistry;
 import org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodePath;
 import org.apache.shardingsphere.elasticjob.lite.util.env.IpUtils;
 
+import java.util.Objects;
 import java.util.regex.Pattern;
 
 /**
@@ -58,7 +60,11 @@ public final class ServerNode {
      * @return is server path for localhost or not
      */
     public boolean isLocalServerPath(final String path) {
-        return path.equals(jobNodePath.getFullPath(String.format(SERVERS, JobRegistry.getInstance().getJobInstance(jobName).getIp())));
+        JobInstance jobInstance = JobRegistry.getInstance().getJobInstance(jobName);
+        if (Objects.isNull(jobInstance)) {
+            return false;
+        }
+        return path.equals(jobNodePath.getFullPath(String.format(SERVERS, jobInstance.getIp())));
     }
     
     String getServerNode(final String ip) {

